src/otx/core/engine/__init__.py

```diff
@@ -0,0 +1,54 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+"""Module for OTX engine components."""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from otx.core.config import TrainConfig
+from otx.core.engine.train import train
+
+if TYPE_CHECKING:
+    from lightning import Trainer
+
+
+class Engine:
+    """OTX Engine class."""
+
+    def __init__(self):
+        self._trainer = None
+
+    @property
+    def trainer(self) -> Trainer:
+        """PyTorch Lightning Trainer.
+
+        To get this property, you should execute `Engine.train()` function first.
+        """
+        if self._trainer is None:
+            msg = "Please run train() first"
+            raise RuntimeError(msg)
+
+        return self._trainer
+
+    def train(self, cfg: TrainConfig) -> dict[str, Any]:
```

Comment:
This is for future reference (not consider it should be done in this phase), however, want to hear opinions.

At the engine level, we need to consider the API. So, `Engine.train(cfg: TrainConfig)` is insufficient.  We need to provide more APIs to have OTX better usage. One of the examples could be `Engine.train(model: nn.Model, data_loader: torch.DataLoader, cfg: TrainConfig)`.

Reply:
For both your comments here and https://github.com/openvinotoolkit/training_extensions/pull/2670#pullrequestreview-1749720718, I think that those are not important to delay our work at this moment. This PR is prepared to let you guys - execution members - start working on the P1 targets (Geti support model migration). What I feel is that there is no definite evidence that can make the future work harder by this design even in your comments. Please start your P1 target and if you find anything that's actually hindering your work, feel free to report it. It is definitely true that we should fix it if there are some flaws. However, we don't know until actually doing it. I think that is how we can work efficiently as a team.

Reply:
Sure, I've been doing the P1 target by using the current branch and have no objection to use this branch for the P1. The reason why I left these comments is we also need to consider the after P1. If the objective of this PR is just for the PoC, I don't mind using AS-IS. Then, we need to further PR for the productization.